### ISSUE TITLE
nrunner tests with TestSuite initialization not work

### ISSUE THREAD

**Issue by richtja** [avocado-framework/avocado#4346]
example:
```
#!/usr/bin/env python3

import sys

from avocado.core.job import Job
from avocado.core.suite import TestSuite
from avocado.core.nrunner import Runnable, Task

config = {
    'run.test_runner': 'nrunner',
}
suite = TestSuite(name="true", config=config, tests=[Task("task1", Runnable('test', '/bin/true'))])


with Job(config, [suite]) as j:
    sys.exit(j.run())

```
output:
```
python /tmp/test.py
JOB ID     : e6f706551ce4c88426da1f83272c66ae638bc6ab
JOB LOG    : /home/jarichte/avocado/job-results/job-2020-12-07T16.30-e6f7065/job.log
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/jarichte/avocado/job-results/job-2020-12-07T16.30-e6f7065/results.html
JOB TIME   : 0.37 s
```
the same problem is with example in  `examples/jobs/passjob_custom.py`

I find out that the runner runs the tests but avocado doesn't receive the status.

**Comment by clebergnu** [comment]
@richtja when you're manually creating `Task`s, you need to set their status servers attribute to match the job's status server uri.

**Comment by clebergnu** [comment]
The following seems to work:

```python
#!/usr/bin/env python3

import sys

from avocado.core.job import Job
from avocado.core.suite import TestSuite
from avocado.core.nrunner import Runnable, Task

config = {
    'run.test_runner': 'nrunner',
}
suite = TestSuite(name="true", config=config, tests=[
    Task("task1", Runnable('exec-test', '/bin/true'), ['127.0.0.1:8888'])]
)


with Job(config, [suite]) as j:
    sys.exit(j.run())
```

Of course, improvements can be made here.

**Comment by richtja** [comment]
Ok, @clebergnu thanks for explanation, I was following the `examples/jobs/passjob_custom.py`, so I will fix that.

**Comment by richtja** [comment]
@clebergnu I think that there is another issue, when I fix the `examples/jobs/passjob_custom.py` like this:
```
import sys

from avocado.core.job import Job
from avocado.core.nrunner import Runnable, Task
from avocado.core.suite import TestSuite

config = {'run.test_runner': 'nrunner'}

suite1 = TestSuite(config=config,
                   tests=[Task("task1", Runnable("noop", "noop"), ['127.0.0.1:8888'])], name='suite1')
suite2 = TestSuite(config=config,
                   tests=[Task("task2", Runnable("noop", "noop"), ['127.0.0.1:8888'])], name='suite2')

with Job(config, [suite1, suite2]) as j:
    sys.exit(j.run())
```
The second suite is not run.
```
python examples/jobs/passjob_custom.py
JOB ID     : 722f4adbfacc678121a0725a6cc2c41e561b7697
JOB LOG    : /home/jarichte/avocado/job-results/job-2020-12-07T16.54-722f4ad/job.log
 (suite1-1/1) noop: STARTED
 (suite1-1/1) noop:  PASS (0.00 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/jarichte/avocado/job-results/job-2020-12-07T16.54-722f4ad/results.html
JOB TIME   : 1.02 s
```

**Comment by clebergnu** [comment]
Seems like a real bug!

**Comment by beraldoleal** [comment]
@richtja the second issue, can you please open a specific issue for that? I can try to debug it.

**Comment by richtja** [comment]
> @richtja about the second issue, can you please open a specific issue for that? I can try to debug it.

@beraldoleal  you mean this?
> @clebergnu I think that there is another issue, when I fix the `examples/jobs/passjob_custom.py` like this:
> 
> ```
> import sys
> 
> from avocado.core.job import Job
> from avocado.core.nrunner import Runnable, Task
> from avocado.core.suite import TestSuite
> 
> config = {'run.test_runner': 'nrunner'}
> 
> suite1 = TestSuite(config=config,
>                    tests=[Task("task1", Runnable("noop", "noop"), ['127.0.0.1:8888'])], name='suite1')
> suite2 = TestSuite(config=config,
>                    tests=[Task("task2", Runnable("noop", "noop"), ['127.0.0.1:8888'])], name='suite2')
> 
> with Job(config, [suite1, suite2]) as j:
>     sys.exit(j.run())
> ```
> 
> The second suite is not run.
> 
> ```
> python examples/jobs/passjob_custom.py
> JOB ID     : 722f4adbfacc678121a0725a6cc2c41e561b7697
> JOB LOG    : /home/jarichte/avocado/job-results/job-2020-12-07T16.54-722f4ad/job.log
>  (suite1-1/1) noop: STARTED
>  (suite1-1/1) noop:  PASS (0.00 s)
> RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
> JOB HTML   : /home/jarichte/avocado/job-results/job-2020-12-07T16.54-722f4ad/results.html
> JOB TIME   : 1.02 s
> ``

Sorry, I expressed myself in a wrong way and this is already solved in #4351
